### DISCUSSION TITLE
Tips for logging

### DISCUSSION THREAD

**Discussion by toadslop** [jonaskello/eslint-plugin-functional#315] [Ideas]
Add tips for avoiding errors when logging.

Every application requires some kind of logging. I think that’s a given. This linter, while excellent, always highlights any logging statement as an error. Though it is true that logging does violate a principle of functional programming in a technical sense, no program actually forgoes logging altogether — they just use functional constructs to wrap their logging statements, like the IO monad. However, this longer provides no feedback for how to log properly, and not only that, if you try to write an IO monad it’ll throw tons of errors telling you that your monad isn’t functional. A linter-approved logging system would be helpful as well as details for how to implement it.

**Comment by RebeccaStevens** [comment]
If anyone would like to provide tips; please do so.
